homeworks/week4/trang_vu

```diff
@@ -0,0 +1,83 @@
+import java.util.*;
+
+class trang_vu{
+  public static void main(String[] args) {
+    Interval[] intervals = {
+      new Interval(1,3),
+      new Interval(2,5),
+      new Interval(1,4),
+      new Interval(0,2)
+    };
+
+    int[] count = new int[5];
+    int iterations = 100000;
+    for (int i = 0; i < iterations; i++) {
+      double f = generateRandomValue(intervals);
+      for (int j = 1; j <= 5; j++)
+      if (f >= j-1 && f < j) {
+        count[j-1] ++;
+      }
+    }
+
+    for (int j = 1; j <= 5; j++) {
+      System.out.println("Interval [" + (j-1) + "," + j + ") with probability " + 100*(double)count[j-1]/iterations + " %");
+    }
+
+    // Interval [0,1) with probability ~10%
+    // Interval [1,2) with probability ~30%
+    // Interval [2,3) with probability ~30%
+    // Interval [3,4) with probability ~20%
+    // Interval [4,5) with probability ~10%
+  }
+
+
+  //utilize function
+  static int arrange(double value, double arr[]) {
+    int left = 0;
+    int right = arr.length -1;
+    while (left < right) {
+      int mid = left + (right - left)/2;
+      if (arr[mid] < value) {
+        left = mid + 1;
+      } else {
+        right = mid;
+      }
+    }
+    return left;
+  }
+
+  // TO DO:
+  // Time complexity 0(logn) : time for sorting/arranging
+  // Space complexity 0(n) : in place
```

Comment:
I am not sure what do you mean by "in place" here.

This solution is very similar to my sample solution.
Note: logN is the binary search time, used to find the index of a certain range.

Please see a simpler solution and see the trade-off in term of space/time.
https://github.com/P2PLearning/DSA/pull/28/files